README.md

```diff
@@ -42,6 +43,32 @@ assert_eq!(i2c.get_last_address(), Some(42));
 assert_eq!(i2c.get_write_data(), &[1, 2, 4]);
 ```
 
+### SPI
+
+```rust
+use hal::blocking::spi::{Transfer, Write};
+use embedded_hal_mock::SPIMock;
+
+let mut spi = SPIMock::new();
+
+// Configure expectations
+spi.expect(vec![
+    Transaction::write(vec![1u8, 2u8]),
+    Transaction::transfer(vec![3u8, 4u8], vec![5u8, 6u8]),
+]);
+
+// Writing
+spi.write(&vec![1u8, 2u8]).unwrap();
+
+// Transferring
+let mut v = vec![3u8, 4u8];
+spi.transfer(&mut v).unwrap();
+assert_eq!(v, vec![5u8, 6u8]);
```

Comment:
Last change request (sorry): I'd call this variable `buf` to indicate that it's a buffer and not "just" a data vector :slightly_smiling_face:

After that I think it should be good to merge.

Reply:
No worries, and done. Are you attached to exposing `SPIMock` at the top level? Because it might be nicer to have `spi::Mock` and `spi::Transaction` imo.

I could rename them then re-export with different names actually, maybe that's better.